synapse/module_api/__init__.py

```diff
@@ -425,6 +425,15 @@ def register_user(
             )
         )
 
+    def get_user_total_message_count(self, user_id):
+        """Counts the total number of messages, sent by a user.
+        Args:
+            user_id (str): full canonical @user:id
+        Returns:
+            defer.Deferred[int] : The number of messages.
+        """
+        return defer.ensureDeferred(self._store.count_total_messages(user_id))
```

Comment:
Could you use Python's async/await semantics here rather than Deferreds?
It's true some other methods use Deferreds, but the reason for that is mostly that we haven't converted them to async/await yet.